src/compiler/checker.ts

```diff
@@ -4796,6 +4796,10 @@ namespace ts {
                 return addOptionality(declaredType, isOptional);
             }
 
+            if (isNamelessJSDocTypeDef(declaration)) {
+                return getDeclaredTypeOfTypeAlias(getSymbolOfNode(declaration));

```

Comment:
so the type of the variable declaration is the type of the typedef? that seems like it basically treats `@typedef` as if it were `@type` and then relies on Javascript's ability to resolve values as types (in resolveEntityName) to then use the variable as a type.

Doesn't that mean this will cause importing the typedef from Typescript to now fail? Can you add a test with a TS file that imports a nameless typedef?

The right way is probably to change resolveEntityName to return the symbol for the variable even when resolving a type (like the current values-as-types code), then make sure that symbol is found correctly through getDeclaredTypeOfSymbol. The symbol might be fine as-is, or might need to be marked with SymbolFlags.TypeAlias.

Reply:
I wanted to make sure nameless typedef's didn't throw an implicit any error and assigning the underlying type seemed acceptable at the time.

Importing the typedef takes a slightly different path when resolving the symbol, where it was incorrectly resolving the the value symbol in `combineValueAndTypeSymbols`. I'll push a fix to that and updated tests shortly.